js-foundations.md

```diff
@@ -15,115 +15,66 @@ logIt()
+  Your answer:
 
-  Researched answer:  JavaScript Object Notation! JSON data is written as name/value pairs, just like JavaScript object properties.
-  A name/value pair consists of a field name (in double quotes), followed by a colon, followed by a value:
-  
-  It is a fomat used of storing and transporting data.
-  
-  Within our class, we have used it mostly to store names: dates: type: location: etc...
-  to maniputlate the final code when inputting different info without rewriting everything. 
+  Researched answer:
 
 
 3. What does CRUD stand for?
 
-  Your answer: functions a website should do (4 important ones) 1. Create 2.read 3. U... 4. delete
+  Your answer:
 
-  Researched answer: From class notes.
-  
-The basic actions a website should be able to do:
-C - Create
-R - Read
-U - Update
-D - Delete
+  Researched answer:
 
 
 
 4. What does are the 5 HTTP verbs?
 
-  Your answer: I don't know. I just know they go hand and hand with CRUD.
+  Your answer:
 
   Researched answer:
-  The HTTP verbs comprise a major portion of our “uniform interface”
-  constraint and provide us the action counterpart to the noun-based resource. 
-  
-Get - read, simplest type of request
-Post - create, usually filling out a form
-Put - update/replace
-Patch - update/modify
-Delete
-HTTP transfers info as strings
-
-
 
 
 5. What is a higher-order function?
 
-  Your answer: A function that takes in another function. 
+  Your answer:
+
+  Researched answer:
 
-  Researched answer: A higher-order function is simply a function that either takes in another function 
-  as an argument or returns another function.
-  
-  .map()
-  .filter()
-  
 
 6. STRETCH: What is a closure, what is it good for and how do you recognize one?
 
-  Your answer:no idea
+  Your answer:
 
   Researched answer:
-  A closure is the combination of a function bundled together (enclosed) with references to
-  its surrounding state (the lexical environment). In other words, a closure gives you access to an
-  outer function’s scope from an inner function. 
-  In JavaScript, closures are created every time a function is created, at function creation time.
-  
-  
-  from my understanding, it was a function within a function 
-  who's inner function could access the orginal function's purpose? 
 
 
 7. STRETCH: What is an API?
 
-  Your answer: ... 
+  Your answer:
 
   Researched answer:
-Application Programs Interface, the process of getting data from a website that is returned 
-in JSON An API is a data to data interface, does not have any markup, just generates information
-Websites have pages, APIs have endpoints
-
-
 
 
 ### Additional Feedback Questions.
 
 1. Do you have a suggestion for a Check In question?
 
-No, I did like this assesmment. We don't review terminology of this sort often and honesly I didn't know most of 
-the questions above. Some I knew how to do without even knowing the name of what i was doing. example, JSON. 
 
```

Comment:
I'm glad you find this helpful. I made flashcards for vocabulary when I was in bootcamp.